docs/CONTRIBUTING.md

```diff
@@ -1,21 +1,278 @@
+* Reviewing a pull request that was submitted by someone else, to check if the feature is correctly authored.
 
-* Go to [feature guidelines](./README.md#feature-guidelines).
-* Go to [open issues](https://github.com/web-platform-dx/web-features/issues) to propose new feature groups or report a bug.
-* Go to [open PRs](https://github.com/web-platform-dx/web-features/pulls) to review incoming feature groups and updates.
+In any case, **thank you** for wanting to help. This document will guide you through the process of contributing to the web-features repository.
 
-Also consider joining [the WebDX Community Group](https://www.w3.org/community/webdx/).
+Also consider joining the [WebDX Community Group](https://www.w3.org/community/webdx/).
+
+## What makes a feature
+
+Think of features as individual components of the web platform that web developers can use to achieve specific tasks.
+
+As such, a feature has no specific size. Some features might cover single CSS properties while other features cover entire JavaScript APIs with multiple interfaces, methods, and properties. As long as a web developer would use this to achieve a specific goal, it's a feature.
+
+For example, the `fetch()` API is a feature, the `:has()` CSS pseudo-class function too, and so is the Web Audio API.
+
+For the time being, this repository focuses only on features that have at least some level of browser support and that are already documented on MDN.
+
+### Feature file name and format
+
+Features in this repository are authored as **YAML files** that are stored in the [`features`](../features) directory. Each file corresponds to a single feature, and contains metadata about the feature, such as its name, description, specification, or support data.
+
+For example, the `fetch()` API feature is described in the [`features/fetch.yml`](https://github.com/web-platform-dx/web-features/blob/main/features/fetch.yml) file.
+
+The name of the file also constitutes the unique ID of the feature. In the example above, the unique ID of the feature is `fetch`.
+
+Feature IDs must be unique as they are used in other projects to reference the feature. For example, the [browser-compat-data](https://github.com/mdn/browser-compat-data/) project references feature IDs in their compatibility data.
+
+### Fields in a feature file
+
+This table lists the fields that can be found in a feature file, and provides a brief description of each field:
+
+| Field | Description | Type | Mandatory |
+|---|---|---|---|
+| `name` | The name of the feature. | String | Yes |
+| `description` | A short description of the feature. | String | Yes |
+| `spec` | One or more specification URLs for this feature | String, or array of strings | Yes |
+| `group` | An optional group, or list of groups that this feature belongs to | String, or array of strings | No |
+| `snapshot` | An optional snapshot, or list of snapshots that this feature belongs to | String, or array of strings | No |
```

Comment:
ditto as for `group`

Reply:
Removed this row for now, as we've not really used snapshots so far.